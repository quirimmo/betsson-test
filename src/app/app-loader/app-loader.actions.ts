import { Injectable } from '@angular/core';
import { Action } from 'redux';
import { NgRedux } from '@angular-redux/store';
import { StoreModel } from '../store/store.model';

export interface AppLoaderAction extends Action {
	isLoading: boolean;
}

@Injectable()
export class AppLoaderActions {
	static readonly START_LOADING = 'START_LOADING';
	static readonly STOP_LOADING = 'STOP_LOADING';

	constructor(private ngRedux: NgRedux<StoreModel>) {}

	public startLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.START_LOADING,
			isLoading: true
		});
	}

	public stopLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.STOP_LOADING,
			isLoading: false
		});
	}
}
