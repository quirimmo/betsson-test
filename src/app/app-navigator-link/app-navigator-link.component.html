<nav>
	<a class="back-navigation-wrapper hvr-sweep-to-right" routerLink="/">
		<mat-icon class="back-navigation-arrow">arrow_left</mat-icon>
		<span class="back-navigation-text">All Movies</span>
	</a>
</nav>
