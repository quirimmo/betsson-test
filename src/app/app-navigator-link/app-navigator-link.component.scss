@import './../../colors.scss';
.back-navigation-wrapper {
	color: $blue;
	position: relative;
	text-transform: uppercase;
	font-size: 15pt;
	letter-spacing: 0.5px;
	top: -20px;
	left: -50px;
	display: block;
	clear: both;
	height: 110px;
	width: 0px;
	opacity: 0.75;
	-webkit-transition: all 0.35s;
	transition: all 0.35s;
}

.back-navigation-wrapper:hover {
	width: 250px;
	border-bottom: 1px solid white;
	opacity: 1;
	color: white;
}

.back-navigation-arrow {
	font-size: 100pt;
	position: absolute;
	display: inline-block;
	top: 0;
	left: 0;
}

.back-navigation-text {
	position: absolute;
	width: 150px;
	text-align: left;
	top: 50px;
	left: 100px;
}

.hvr-sweep-to-right:before {
	background: $blue;
}
