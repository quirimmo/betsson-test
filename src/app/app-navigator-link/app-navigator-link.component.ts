import { Component } from '@angular/core';
import { MoviesRouterService } from '../movies/movies-router.service';

@Component({
	selector: 'app-navigator-link',
	templateUrl: './app-navigator-link.component.html',
	styleUrls: ['./app-navigator-link.component.scss']
})
export class AppNavigatorLinkComponent {
	constructor(private moviesRouterService: MoviesRouterService) {}

	public navigateBackToMoviesList(): void {
		this.moviesRouterService.navigateBackToMoviesList();
	}
}
