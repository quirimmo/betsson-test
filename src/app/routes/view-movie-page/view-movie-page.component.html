<div>
	<a routerLink="/">See all the movies</a>
	<movie-list-item [movie]="movie"></movie-list-item>
</div>
