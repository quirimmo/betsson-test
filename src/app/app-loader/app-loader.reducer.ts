import { AppLoaderAction, AppLoaderActions } from './app-loader.actions';

export function appLoaderReducer(
	state: boolean = false,
	action: AppLoaderAction
): boolean {
	switch (action.type) {
		case AppLoaderActions.START_LOADING:
			return action.isLoading;
		case AppLoaderActions.STOP_LOADING:
			return action.isLoading;
		default:
			return state;
	}
}
