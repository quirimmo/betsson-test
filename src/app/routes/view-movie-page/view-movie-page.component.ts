import { Component, OnInit, AfterViewInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { AppLoaderActions } from '../../app-loader/app-loader.actions';
import Movie from 'src/app/movies/movie.model';

@Component({
	selector: 'view-movie-page',
	inputs: [],
	outputs: [],
	templateUrl: './view-movie-page.component.html'
})
export class ViewMoviePageComponent implements OnInit, AfterViewInit {
	movie: Movie;

	constructor(
		private router: Router,
		private route: ActivatedRoute,
		private appLoaderActions: AppLoaderActions
	) {}

	ngOnInit(): void {
		this.movie = this.route.snapshot.data.movie;
		if (!this.movie) {
			this.router.navigate(['']);
		}
	}

	ngAfterViewInit(): void {
		this.appLoaderActions.stopLoading();
	}
}
