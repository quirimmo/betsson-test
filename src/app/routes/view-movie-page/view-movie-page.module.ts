import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ViewMoviePageComponent } from './view-movie-page.component';
import { MoviesModule } from 'src/app/movies/movies.module';
import { RouterModule } from '@angular/router';

@NgModule({
	declarations: [ViewMoviePageComponent],
	imports: [CommonModule, MoviesModule, RouterModule],
	providers: [ViewMoviePageComponent],
	bootstrap: []
})
export class ViewMoviePageRouteModule {}
