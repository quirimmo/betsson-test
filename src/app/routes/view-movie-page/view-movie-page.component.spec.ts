import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppLoaderActions } from 'src/app/app-loader/app-loader.actions';
import { ActivatedRoute, Router } from '@angular/router';
import { ViewMoviePageComponent } from './view-movie-page.component';
import { MovieDetailsComponent } from 'src/app/movies/movie-details/movie-details.component';
import { AppNavigatorLinkComponent } from 'src/app/app-navigator-link/app-navigator-link.component';

const mockStopLoading = jest.fn();
const mockAppLoaderActions = {
	provide: AppLoaderActions,
	useValue: {
		stopLoading: mockStopLoading
	}
};
const mockActivatedRoute = {
	provide: ActivatedRoute,
	useValue: {
		snapshot: {
			data: { movie: 'movie 1' }
		}
	}
};
const mockNavigate = jest.fn();
const mockRouter = {
	provide: Router,
	useValue: {
		navigate: mockNavigate
	}
};
const providers = [mockAppLoaderActions, mockActivatedRoute, mockRouter];
const imports = [RouterTestingModule];
const declarations = [
	ViewMoviePageComponent,
	MovieDetailsComponent,
	AppNavigatorLinkComponent
];

const MockMovieDetailsComponent = {
	set: {
		inputs: ['movie'],
		template: '<span>movie details</span>'
	}
};
const MockAppNavigatorLinkComponent = {
	set: {
		template: '<span>app navigator link</span>'
	}
};

describe('ViewMoviePage Component', () => {
	let componentFixture, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		})
			.overrideComponent(
				AppNavigatorLinkComponent,
				MockAppNavigatorLinkComponent
			)
			.overrideComponent(MovieDetailsComponent, MockMovieDetailsComponent)
			.compileComponents();
		componentFixture = TestBed.createComponent(ViewMoviePageComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
	}));

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});

	describe('OnInit', () => {
		beforeEach(async () => {
			componentFixture.detectChanges();
			await componentFixture.whenStable();
		});
		it('should init the movie', () => {
			expect(componentInstance.movie).toEqual('movie 1');
		});

		it('should dispatch the stopLoading action', () => {
			expect(mockStopLoading).toHaveBeenCalled();
		});
	});
});
