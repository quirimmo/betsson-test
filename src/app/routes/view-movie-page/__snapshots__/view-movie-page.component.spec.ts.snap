// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMoviePage Component should match the snapshot 1`] = `
<view-movie-page
  appLoaderActions={[Function Object]}
  movie={[Function String]}
  route={[Function Object]}
  router={[Function Object]}
>
  <section
    class="container-fluid"
  >
    <div>
      <app-navigator-link>
        <span>
          app navigator link
        </span>
      </app-navigator-link>
    </div>
    <div>
      <movie-details>
        <span>
          movie details
        </span>
      </movie-details>
    </div>
  </section>
</view-movie-page>
`;
